syntax = "proto3";

package com.vsu001.ethernet.core;

option java_multiple_files = true;
option java_package = "com.vsu001.ethernet.core.service";
option java_outer_classname = "ConfigServiceProto";

service ConfigService {
  // Get Neo4j connection configurations
  rpc GetNeo4jServingConfig(GetNeo4jServingConfigRequest) returns (GetNeo4jServingConfigResponse);

  // Get serving connection configurations
  rpc GetServingConfig(GetServingConfigRequest) returns (GetServingConfigResponse);
}

message GetNeo4jServingConfigRequest {}

message GetNeo4jServingConfigResponse {
  // Neo4j connection URI
  string connection_uri = 1;

  // Neo4j username
  string username = 2;

  // Neo4j password
  string password = 3;

}

message GetServingConfigRequest {}

message GetServingConfigResponse {
  // rpyc host
  string rpyc_host = 1;

  // rypc port
  string rpyc_port = 2;

  // Docker container name running neo4j
  string docker_container_name = 3;
}